cern_access/indico_cern_access/plugin.py

```diff
@@ -0,0 +1,237 @@
+
+    @staticmethod
+    def from_python(value):
+        return value.id
+
+    @staticmethod
+    def to_python(value):
+        return DesignerTemplate.get(value)
+
+
+class CERNAccessPlugin(IndicoPlugin):
+    """CERN Access Request
+
+    Provides a service request through which event managers can ask for
+    access to the CERN site for the participants of the event
+    """
+
+    settings_form = PluginSettingsForm
+    configurable = True
+    default_settings = {
+        'adams_url': '',
+        'login': '',
+        'password': '',
+        'secret_key': '',
+        'access_ticket_template_id': None,
+        'excluded_categories': []
+    }
+    settings_converters = {
+        'access_ticket_template_id': DesignerTemplateConverter
+    }
+    acl_settings = {'authorized_users'}
+
+    def init(self):
+        super(CERNAccessPlugin, self).init()
+        self.template_hook('registration-status-flag', self._get_access_status)
+        self.template_hook('registration-status-action-button', self._get_access_action_button)
+        self.connect(signals.plugin.get_event_request_definitions, self._get_event_request_definitions)
+        self.connect(signals.event.registration_deleted, self._registration_deleted)
+        self.connect(signals.event.timetable.times_changed, self._event_time_changed, sender=Event)
+        self.connect(signals.event.registration_personal_data_modified, self._registration_modified)
+        self.connect(signals.event.registration_form_deleted, self._registration_form_deleted)
+        self.connect(signals.event.deleted, self._event_deleted)
+        self.connect(signals.event.updated, self._event_title_changed)
+        self.connect(signals.event.is_ticketing_handled, self._is_ticketing_handled)
+        self.connect(signals.form_validated, self._form_validated)
+        self.connect(signals.event.designer.print_badge_template, self._print_badge_template)
+        self.connect(signals.event.registration.generate_ticket_qr_code, self._generate_ticket_qr_code)
+
+    def get_blueprints(self):
+        yield blueprint
+
+    def _get_event_request_definitions(self, sender, **kwargs):
+        return CERNAccessRequestDefinition
+
+    def _get_access_action_button(self, regform):
+        if regform.cern_access_request and regform.cern_access_request.is_active:
+            return render_plugin_template('cern_access_action_button.html', regform=regform)
+
+    def _get_access_status(self, regform, registration=None, header=False):
+        if regform.cern_access_request and regform.cern_access_request.is_active:
+            return render_plugin_template('cern_access_status.html',
+                                          registration=registration,
+                                          header=header)
+
+    def _registration_deleted(self, registration, **kwargs):
+        """Withdraws CERN access request for deleted registrations"""
+        if registration.cern_access_request:
```

Comment:
what if the access request is `withdrawn` or `not_requested`? do we really need to make an adams API call in that case?